quill-core/src/main/scala/io/getquill/ast/AstOps.scala

```diff
@@ -0,0 +1,81 @@
+package io.getquill.ast
+
+import io.getquill.norm.BetaReduction
+
+object Implicits {
+  implicit class AstOpsExt(body: Ast) {
+    def reduce(reductions: (Ast, Ast)*) = BetaReduction(body, reductions: _*)
+    def +||+(other: Ast) = BinaryOperation(body, BooleanOperator.`||`, other)
+    def +&&+(other: Ast) = BinaryOperation(body, BooleanOperator.`&&`, other)
+    def +==+(other: Ast) = BinaryOperation(body, EqualityOperator.`==`, other)
+  }
+}
+
+object +||+ {
+  def unapply(a: Ast): Option[(Ast, Ast)] = {
+    a match {
+      case BinaryOperation(one, BooleanOperator.`||`, two) => Some((one, two))
+      case _ => None
+    }
+  }
+}
+
+object +&&+ {
+  def unapply(a: Ast): Option[(Ast, Ast)] = {
+    a match {
+      case BinaryOperation(one, BooleanOperator.`&&`, two) => Some((one, two))
+      case _ => None
+    }
+  }
+}
+
+object +==+ {
+  def unapply(a: Ast): Option[(Ast, Ast)] = {
+    a match {
+      case BinaryOperation(one, EqualityOperator.`==`, two) => Some((one, two))
+      case _ => None
+    }
+  }
+}
+
+object Exist {
```

Comment:
Wdyt about renaming these to `IsNotNullCheck`/`isNullCheck`/etc

Reply:
NP. Will do.

Reply:
Come to think of it, how about just `IsNotNull`/`isNotNull`? Also `Exist` would become `IsNull`/`isNull`.

Reply:
It might be confusing as well. If I see `IsNull(ast)`, it gives me the impression that it checks if the ast is null or the AST contains a null literal constant

Reply:
I don't feel that strongly about it either. We can go with `IsNotNullCheck`